src/domains/user/services/userService.js

```diff
@@ -0,0 +1,20 @@
+import { db } from "@/shared/lib/firebase/config";
+import { doc, getDoc, } from "firebase/firestore";
+
+
+const getUserProfile = async (uid) => {
+  // 1. "users" 컬렉션에서 uid 문서를 가리키는 docRef를 만든다.
+  // 2. getDoc(docRef) 실행 → 결과를 snap 이라고 하자.
+  // 3. snap.exists() 이면 → snap.data() 반환
+  //    아니라면 → null 반환
+  const snap = await getDoc(doc(db, "users", uid));
+  if(snap.exists()){
+    return snap.data();
+  }else{
+    return null;
+  }
+  // (await getDoc(doc(db, "users", uid))).data() ?? null;
+}
```

Comment:
```suggestion
export const getUserProfile = async (uid) => {
  try {
    if(snap.exists()){
      return snap.data();
    }else{
      return null;
     }
  } catch (error) {
    console.error("Error fetching user profile:", error);
    return null;
  }
};
```

추후 에러 핸들링을 위해 서비스 코드는 try~catch 사용하시는 것을 권장드립니다!